packages/reactive-element/src/test/decorators/queryAssignedElements_test.ts

```diff
@@ -0,0 +1,162 @@
+  // the decorator can be tested.
+  @customElement(generateElementName())
+  class C extends RenderingElement {
+    div!: HTMLDivElement;
+    div2!: HTMLDivElement;
+    div3!: HTMLDivElement;
+    assignedEls!: D;
+    assignedEls2!: E;
+    @queryAssignedElements() missingSlotAssignedElements!: Element[];
+
+    override render() {
+      return html`
+        <assigned-elements-el
+          ><div id="div1">A</div>
+          <slot slot="footer"></slot
+        ></assigned-elements-el>
+        <assigned-elements-el-2><div id="div2">B</div></assigned-elements-el-2>
+      `;
+    }
+
+    override firstUpdated() {
+      this.div = this.renderRoot.querySelector('#div1') as HTMLDivElement;
+      this.div2 = this.renderRoot.querySelector('#div2') as HTMLDivElement;
+      this.div3 = this.renderRoot.querySelector('#div3') as HTMLDivElement;
+      this.assignedEls = this.renderRoot.querySelector(
+        'assigned-elements-el'
+      ) as D;
+      this.assignedEls2 = this.renderRoot.querySelector(
+        'assigned-elements-el-2'
+      ) as E;
+    }
+  }
+
+  setup(async () => {
+    container = document.createElement('div');
+    document.body.append(container);
+    el = new C();
+    container.append(el);
+    await new Promise((r) => setTimeout(r, 0));
+  });
+
+  teardown(() => {
+    container?.remove();
+  });
+
+  test('returns assignedElements for slot', () => {
+    // Note, `defaultAssigned` does not `flatten` so we test that the property
+    // reflects current state and state when nodes are added or removed.
+    assert.deepEqual(el.assignedEls.defaultAssigned, [el.div]);
+    const child = document.createElement('div');
+    const text1 = document.createTextNode('');
+    el.assignedEls.append(text1, child);
+    const text2 = document.createTextNode('');
+    el.assignedEls.append(text2);
+    flush();
+    assert.deepEqual(el.assignedEls.defaultAssigned, [el.div, child]);
+    child.remove();
+    flush();
+    assert.deepEqual(el.assignedEls.defaultAssigned, [el.div]);
+  });
+
+  test('returns assignedElements for unnamed slot that is not first slot', () => {
+    assert.deepEqual(el.assignedEls2[defaultSymbol], [el.div2]);
+  });
+
+  test('returns flattened assignedElements for slot', () => {
+    assert.deepEqual(el.assignedEls.footerAssigned, []);
+    const child1 = document.createElement('div');
+    const child2 = document.createElement('div');
+    el.append(child1, child2);
+    flush();
+    assert.deepEqual(el.assignedEls.footerAssigned, [child1, child2]);
+
+    assert.equal(el.assignedEls.footerNotFlattenedSlot.length, 1);
+    assert.equal(el.assignedEls.footerNotFlattenedSlot?.[0]?.tagName, 'SLOT');
+
+    child2.remove();
+    flush();
+    assert.deepEqual(el.assignedEls.footerAssigned, [child1]);
+  });
+
+  test('always returns an array, even if the slot is not rendered', () => {
+    assert.isArray(el.missingSlotAssignedElements);
```

Comment:
I'm surprised that this test was here before the change to always return an array. Was it failing? If it was passing before... why?

Reply:
It was passing because it was selecting: `<slot slot="footer"></slot>`.